metadata-ingestion/setup.py

```diff
@@ -223,7 +223,7 @@
 
 iceberg_common = {
     # Iceberg Python SDK
-    "pyiceberg",
+    "pyiceberg~=0.4",
```

Comment:
```suggestion
    "pyiceberg>=0.4",
```
Pretty sure this limits us to between 0.4 and 0.5

Reply:
```
For a given release identifier V.N, the compatible release clause is approximately equivalent to the pair of comparison clauses:

>= V.N, == V.*
```

Reply:
Ahh cause major is always 0, sounds good